src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderAdapter.cs

```diff
@@ -46,13 +48,27 @@ public VsCredentialProviderAdapter(IVsCredentialProvider provider)
             }
 
             // TODO: Extend the IVS API surface area to pass down the credential request type.
-            var credentials = await _provider.GetCredentialsAsync(
+
+            // Telling the credential provider to cancel the request in situations like PM UI being closed or the search query changing
+            // is disruptive, because it may prompt the customer for interactive input, but then not get/cache a token, so the next request
```

Comment:
I imagine we're ok with NuGet not caching the acquired creds, but only the cred provider. 

Then later when NuGet actually needs the cred, it'll hit the cred provider but we'd get creds back with no user prompt?

Reply:
yes, that's exactly the goal 👍 

Should I improve the comment to make this more obvious?

Reply:
up  to you. 

You can add the part about NuGet caching the tokens, but that's only important because of I know how the NuGet side works.